### PR TITLE
feat: 🎸 Added DotLottie loader. and updated constructor method

### PR DESCRIPTION
**Pull request by afsalz**

### CODE REVIEW

**Review by theashraf — APPROVED** [review]

File: dotlottie/src/main/java/com/lottiefiles/dotlottie/core/loader/NetworkLoader.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                .build()
+
+            val response = httpClient.newCall(request).execute()
+
+            if (!response.isSuccessful) {
+                throw IOException("[NetworkLoader]: Failed to download file: $url")
+            }
+
+            val isJson =
+                response.header("Content-Type", "")?.contains("application/json") ?: false
```
[the remark this answers is not in the document]
Comment (theashraf) [reply dotlottie/src/main/java/com/lottiefiles/dotlottie/core/loader/NetworkLoader.kt:78]: We can assume it's lottie, and the core kotlin bindings is going to throw onLoadError if it's not a valid lottie json

**Review by theashraf — COMMENTED** [review]
Awesome 🚀

File: dotlottie/src/main/java/com/lottiefiles/dotlottie/core/loader/NetworkLoader.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (afsalz) [reply dotlottie/src/main/java/com/lottiefiles/dotlottie/core/loader/NetworkLoader.kt:78]: Cool. will add this in

File: dotlottie/src/main/java/com/lottiefiles/dotlottie/core/compose/runtime/DotLottieController.kt [hunk 2]
```diff
@@ -195,6 +196,20 @@ class DotLottieController {
         }
     }
 
+    fun loadAnimation(
+        animationId: String,
+        width: UInt = this.width.value,
+        height: UInt = this.height.value
```
[the remark this answers is not in the document]
Comment (afsalz) [reply dotlottie/src/main/java/com/lottiefiles/dotlottie/core/compose/runtime/DotLottieController.kt:202]: this is optional. users can ignore this

**Pull request merged**